Deleting Duplicate Messages

JacobMGEvans opened the pull request:
- [x] Find messages based on duplicate content & by the same author

- [x] If the message is within 10 secs

- [ ] Creates a message in Bot Channel to the author of duplicate message notifying them of deletion and reasoning

- [ ] Test these new behaviors work in Discord 

- [ ] Write Tests for new behavior

kentcdodds commented on src/admin/index.js:
```diff
@@ -0,0 +1,7 @@
+const {dedupeMessages} = require('./deduping-channel-posts')
+
+function setup(client) {
+  dedupeMessages(client)
```
The other utilities interact with the client directly here and put the event handlers in the other files. I like that. Could we do that here as well. It will involve moving the `client.on` call from `dedupeMessages` and simply exporting the message handler? You can get the client from `message.client` :)

JacobMGEvans replied:
I am not entirely certain what you are asking me to do here. Are you just saying convert this admin file into more of a utility and move it into the message index? In other words no Admin?

JacobMGEvans replied:
Ok I think I get it, move the client.on higher up so that other functions can interact with it

kentcdodds replied:
Sorry, it's hard to explain over text 😅 Maybe we can collab on this on Monday.

kentcdodds commented on src/admin/deduping-channel-posts.js:
```diff
@@ -0,0 +1,24 @@
+      )
+    }
+
+    message.channel
```
This will look for other messages within the same channel. We need to look through all the channels' messages. I'm not sure of the best way to accomplish this at the moment. Will need to take some thought...

JacobMGEvans replied:
I checked the documentation and was under the impression is checked all channels, I could have misread that.

kentcdodds replied:
It's possible! I've never heard of `awaitMessages` or "collections" before, so this is helpful!

JacobMGEvans replied:
Yeah I think that the /channel will fire when a message is created in any channel

kentcdodds commented on src/admin/deduping-channel-posts.js:
```diff
@@ -0,0 +1,24 @@
+      .then(collected => collected.last().delete())
+      .catch(console.error)
+
+    client.channels
+      .get('#🤖-talk-to-bots') //! I dont know the Bot Channels ID
```
We'll want to make sure we only do this when a message is actually deleted. Take a look at other files to see how I get a channel by its name. This may be a good time to create a utility for that very purpose.

JacobMGEvans replied:
Not sure how to make a reusable utility for this purpose. I can dig in more for the information.

kentcdodds reviewed (commented):
Thanks for taking this on! I know it must be hard without actually being able to test things out 😅. Here's some early feedback.

kentcdodds commented:
Oh, and let's say if the message is within 30 minutes rather than 10 seconds.

And in our message to them we can tell them that if they put the message in the wrong channel by accident they can delete it in the wrong channel and *then* paste it in the right one.

JacobMGEvans commented:
So the message generation within the dedupe function still needed the client, so that probably tells me that there needs to be some thinking around this. However, I increased the time, moved the message creation to the deletion promise block, and moved the client event higher up.

JacobMGEvans commented:
Monday sounds good to me, I will be technically working and since I took a week off for moving I am sure I will be showing up to millions of emails and whatever unknowns! lol

kentcdodds commented:
This has been merged I promise.
